### PR TITLE
Order item controller refactor

### PR DESCRIPTION
**Pull request by RahulJha-mastek**
Story [AB#13236](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/13236)
Task [AB#14461](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/14461)

### CODE REVIEW

**Review by bjss-chris — APPROVED** [review]

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderItemsControllerTests.cs [hunk 1]
```diff
@@ -47,9 +47,11 @@ public static void Constructors_VerifyGuardClauses()
         [Test]
         [InMemoryDbAutoData]
         public static async Task ListAsync_OrderDoesNotExist_ReturnsNotFound(
+            [Frozen] Mock<IOrderItemService> service,
             CallOffId callOffId,
             OrderItemsController controller)
         {
+            service.Setup(o => o.GetOrderItems(callOffId, It.IsAny<CatalogueItemType?>())).ReturnsAsync((List<OrderItem>)null);
```
Comment (kiran-mastek) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderItemsControllerTests.cs:54]: Should this not be tested for other `CatalogueItemTypes` besides `null` also? 
Suggest using a random value from possible values if agree.

File: tests/NHSD.BuyingCatalogue.Ordering.Services.UnitTests/OrderItemServiceTests.cs [hunk 2]
```diff
@@ -111,14 +111,14 @@ public static void Constructor_NullAccessor_ThrowsArgumentNullException()
 
         [Test]
         [InMemoryDbAutoData]
-        public static async Task GetOrderItems_ReturnsEmpty(
+        public static async Task GetOrderItems_ReturnsNull(
```
Comment (kiran-mastek) [inline tests/NHSD.BuyingCatalogue.Ordering.Services.UnitTests/OrderItemServiceTests.cs:114]: This and other test names in the file do not give a full description of what's being tested so its easier to understand what's going on.
Suggest a format of <MethodName>_<InputOrCondition>_<Result> 
e.g. this title could be `GetOrderItems_InvalidCallOffId_ReturnsNull`.

File: tests/NHSD.BuyingCatalogue.Ordering.Services.UnitTests/OrderItemServiceTests.cs [hunk 3]
```diff
@@ -111,14 +111,14 @@ public static void Constructor_NullAccessor_ThrowsArgumentNullException()
...
             CallOffId callOffId,
             CatalogueItemType catalogueItemType,
             OrderItemService service)
         {
```
Comment (kiran-mastek) [inline tests/NHSD.BuyingCatalogue.Ordering.Services.UnitTests/OrderItemServiceTests.cs:118]: A pre condition/check to prove that the `callOffId` is invalid is recommended to prove that there is no item with the Id in the database.
i.e. `(await context.Order.AnyAsync(o => o.Id == callOffId.Id)).Should().BeFalse();`

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderItemsControllerTests.cs (hunk 1 above)
Comment (RahulJha-mastek) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderItemsControllerTests.cs:54]: With PR [424](https://github.com/nhs-digital-gp-it-futures/BuyingCatalogueOrdering/pull/424/files)
Added one more test case with Non-nullable CatalogueItemTypes

File: tests/NHSD.BuyingCatalogue.Ordering.Services.UnitTests/OrderItemServiceTests.cs (hunk 2 above)
Comment (RahulJha-mastek) [reply tests/NHSD.BuyingCatalogue.Ordering.Services.UnitTests/OrderItemServiceTests.cs:114]: With PR [424](https://github.com/nhs-digital-gp-it-futures/BuyingCatalogueOrdering/pull/424/files)
Renamed unit test method

File: tests/NHSD.BuyingCatalogue.Ordering.Services.UnitTests/OrderItemServiceTests.cs (hunk 3 above)
Comment (RahulJha-mastek) [reply tests/NHSD.BuyingCatalogue.Ordering.Services.UnitTests/OrderItemServiceTests.cs:118]: With PR [424](https://github.com/nhs-digital-gp-it-futures/BuyingCatalogueOrdering/pull/424/files)
Checked existence of Order with invalid CallOfId

**Pull request merged**